pkgs/build-support/setup-hooks/patch-ant-buildfiles.sh

```diff
@@ -0,0 +1,30 @@
+
+
+# Sets the `modificationtime` attribute on the <jar> and <war> ant tasks
+# This is done to make the final archives deterministic
+patchAntBuildfile() {
+    local buildfile="$1"
+    # using `local-name()` ignores the `xmlns` attribute in a parent node
+    @xmlstarlet@ ed --inplace \
+        --var T '//*[local-name()="jar" or local-name()="war"]' \
+        --update '$T/@modificationtime' -v 0 \
+        --insert '$T[not(@modificationtime)]' -t attr -n 'modificationtime' -v 0 \
+        "$buildfile"
+}
+
+patchAntBuildfilesHook() {
+    find "$NIX_BUILD_TOP" -name '*.xml' -type f -writable -print0 |
```

Comment:
Since ant defaults to read `build.xml`, perhaps the hook should only look for those files by default? I realize that you verify the file contents, so there's little risk of editing the wrong files, but it seems a bit "brute force" to me to process all XML files. One alternative is to make the file pattern configurable/overridable per derivation.

(Just thinking out loud, this is not me blocking merge.)

Reply:
If someone has a problem, they can always set `dontPatchAntBuildfiles` and call `patchAntBuildfile` manually. Implementing a pattern based blacklist/whitelist system is a bit too tedious IMO.

Reply:
I wasn't thinking allow/block list, more like filePattern=${antHookPattern:-build.xml} and pass filePattern to find.